modules/benchmarks-cs/ia_loop.cs

```diff
@@ -0,0 +1,372 @@
+        int new_hash = targetable.quad.GetHashCode();
+        targetable.quad = new_hash;
+        GameTargetableState.UpdateByentity_id(entity_id, targetable);
+
+        if (GameLiveTargetableState.FindByentity_id(entity_id) is not null)
+        {
+            GameLiveTargetableState.UpdateByentity_id(entity_id, new(entity_id, new_hash));
+        }
+
+        GameMobileEntityState mobile_entity =
+            GameMobileEntityState.FindByentity_id(entity_id)
+            ?? throw new Exception("GameMobileEntityState Entity ID not found");
+        mobile_entity.location_x += 1;
+        mobile_entity.location_y += 1;
+        mobile_entity.timestamp = MomentMilliseconds();
+
+        GameEnemyAiAgentState.UpdateByentity_id(entity_id, agent);
+
+        GameMobileEntityState.UpdateByentity_id(entity_id, mobile_entity);
+    }
+
+    public static void AgentLoop(
+        GameEnemyAiAgentState agent,
+        GameTargetableState agent_targetable,
+        List<GameTargetableState> surrounding_agents,
+        ulong current_time_ms
+    )
+    {
+        ulong entity_id = agent.entity_id;
+
+        IEnumerable<GameMobileEntityState> coordinates =
+            GameMobileEntityState.FilterByentity_id(entity_id)
+            ?? throw new Exception("GameMobileEntityState Entity ID not found");
+
+        GameEnemyState agent_entity =
+            GameEnemyState.FindByentity_id(entity_id)
+            ?? throw new Exception("GameEnemyState Entity ID not found");
+
+        GameHerdCache agent_herd =
+            GameHerdCache.FindByid(agent_entity.herd_id)
+            ?? throw new Exception("GameHerdCache Entity ID not found");
+
+        SmallHexTile agent_herd_coordinates = agent_herd.location;
+
+        MoveAgent(ref agent, agent_herd_coordinates, current_time_ms);
+    }
+
+    [SpacetimeDB.Reducer]
+    public static void game_loop_enemy_ia(ulong players)
+    {
+        uint count = 0;
+        ulong current_time_ms = MomentMilliseconds();
+
+        foreach (GameEnemyAiAgentState agent in GameEnemyAiAgentState.Iter())
+        {
+            if (agent.next_action_timestamp > current_time_ms)
+            {
+                continue;
+            }
+
+            GameTargetableState agent_targetable =
+                GameTargetableState.FindByentity_id(agent.entity_id)
+                ?? throw new Exception("No TargetableState for AgentState entity");
+
+            List<GameTargetableState> surrounding_agents = GetTargetablesNearQuad(agent_targetable.entity_id, players);
+
+            GameEnemyAiAgentState newAgent = agent with { action = AgentAction.Fighting };
+
+            AgentLoop(newAgent, agent_targetable, surrounding_agents, current_time_ms);
+
+            count++;
+        }
+
+        Runtime.Log($"ENEMY IA LOOP PLAYERS: {players}, processed: {count}");
+    }
+
+    [SpacetimeDB.Reducer]
+    public static void init_game_ia_loop(uint initial_load)
```

Comment:
```suggestion
   // We check only for a single pass in the game loop.
    public static void init_game_ia_loop(uint initial_load)
```

I think is good idea to preserve this comment

Reply:
There are quite a few comments that I didn't copy from Rust version. I just added README to C# modules saying that code is inherited from Rust version, and that Rust version should act as a reference that people should look at.